src/engine/sync/enginesync.cpp

```diff
@@ -110,32 +110,33 @@ void EngineSync::requestEnableSync(Syncable* pSyncable, bool bEnabled) {
             double targetBeatDistance = 0.0;
             double targetBaseBpm = 0.0;
 
-            foreach (const Syncable* other_deck, m_syncables) {
-                if (other_deck == pSyncable) {
+            for (const auto& pOtherSyncable: m_syncables) {
+                if (pOtherSyncable == pSyncable) {
                     // skip this deck
                     continue;
                 }
-                if (!other_deck->getChannel()->isMasterEnabled()) {
+                if (!pOtherSyncable->getChannel()->isMasterEnabled()) {
                     // skip non-master decks, like preview decks.
                     continue;
                 }
 
-                double otherDeckBpm = other_deck->getBpm();
+                double otherDeckBpm = pOtherSyncable->getBpm();
                 if (otherDeckBpm > 0.0) {
-                    // If the requesting deck is playing, but the other deck
-                    // is not, do not sync.
-                    if (pSyncable->isPlaying() && !other_deck->isPlaying()) {
+                    // If the requesting deck is playing, or we have already a
+                    // non plaing deck found, only watch out for playing decks.
+                    if ((foundTargetBpm || pSyncable->isPlaying())
```

Comment:
I think the original logic was right. The bug is that if we go through all the decks and we don't find a syncable playing deck, we end up falling back to the last-seen deck.

I think the correct fix could be below here:
if (pSyncable->isPlaying() && !other_deck->isPlaying()) {
 continue;
}

// Only update the target bpm if we haven't already got one, or we found a playing syncable deck
if (!foundTargetBpm || pOtherSyncable->isPlaying()) {
  foundTargetBpm = true;
  targetBpm = otherDeckBpm;
  targetBaseBpm = pOtherSyncable->getBaseBpm();
  targetBeatDistance = pOtherSyncable->getBeatDistance();
}

if (pOtherSyncable->isPlaying()) {...

Reply:
My Version syncs a non playing deck to the first other non playing deck, if no Deck is playing at all. 
Is it intended to remove this feature? I am unsure, but It may feel like a regression.

Reply:
ok I see how it's changed around.  I think this is fine, thanks!